### PR TITLE
MISUV-8931-CS added new auth solutions to controllers that don't support supporting agents

### PR DESCRIPTION
**Pull request by Chrissystk**

### CODE REVIEW

File: test/controllers/optOut/OptOutCancelledControllerSpec.scala [hunk 1]
```diff
@@ -44,15 +44,15 @@ class OptOutCancelledControllerSpec extends MockAuthActions with MockOptOutServi
...
+  override lazy val app: Application = applicationBuilderWithAuthBindings
     .overrides(
       api.inject.bind[OptOutService].toInstance(mockOptOutService)
     )
     .configure(config)
     .build()
 
-  val testController = fakeApplication().injector.instanceOf[OptOutCancelledController]
-  val optOutCancelledView = fakeApplication().injector.instanceOf[OptOutCancelledView]
+  val testController = app.injector.instanceOf[OptOutCancelledController]
```
Comment (avelx) [inline test/controllers/optOut/OptOutCancelledControllerSpec.scala:54]: might worth to change these two vals to be lazy? as we are using lazy app right away

File: test/controllers/optOut/ConfirmedOptOutControllerSpec.scala [hunk 2]
```diff
@@ -35,12 +35,12 @@ import scala.concurrent.Future
...
   with MockOptOutService {
 
-  override def fakeApplication(): Application = applicationBuilderWithAuthBindings()
+  override lazy val app: Application = applicationBuilderWithAuthBindings
     .overrides(
       api.inject.bind[OptOutService].toInstance(mockOptOutService)
     ).build()
 
-  val testController = fakeApplication().injector.instanceOf[ConfirmedOptOutController]
+  val testController = app.injector.instanceOf[ConfirmedOptOutController]
```
Comment (avelx) [inline test/controllers/optOut/ConfirmedOptOutControllerSpec.scala:43]: same thing, we are using lazy val right away

**Comment by avelx** [comment]
Next pattern looks a bit strange to me as we use lazy val right away:
```
override lazy val app: Application = ... 
val testController = app.injector.instanceOf
```

File: it/test/controllers/CalculationPollingControllerISpec.scala [hunk 3]
```diff
@@ -16,146 +16,245 @@
...
+              "calculation service returns retryable response back initially and then returns success response before interval time completed" in {
+                stubAuthorised(mtdUserRole)
+                IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesAndUkProperty)
+                IncomeTaxCalculationStub.stubGetCalculationErrorResponseByCalcId(testNino, "idFour", testTaxYear)(NO_CONTENT,
+                  LiabilityCalculationError(NO_CONTENT, "no content"))
+
+                val res = buildGETMTDClient(finalCalcPath, additionalCookies ++ Map(SessionKeys.calculationId -> "idFour"))
+
+                //After 1.75 seconds responding with success message
+                Thread.sleep(1750)
```
Comment (avelx) [inline it/test/controllers/CalculationPollingControllerISpec.scala:190]: Please don't use Thread.sleep anywhere in our code base

Comment (Chrissystk) [reply it/test/controllers/CalculationPollingControllerISpec.scala:190]: This is a preexisting tests and not something I introduced. In this case as well they are testing polling so the thread.sleep is delaying the successful stub call not waiting for a result so the thread.sleep approach is not wrong in this scenerio.

File: test/controllers/optOut/ConfirmedOptOutControllerSpec.scala (hunk 2 above)
Comment (Chrissystk) [reply test/controllers/optOut/ConfirmedOptOutControllerSpec.scala]: I have updated to use lazy val but I don’t believe it was necessary as yes the “val” was called straight away but the nature of classes mean that they are only executed at run time. The issue seemed to be more around the fakeApplication() function. I expect under the covers it was being called several times and even though we was stopping the play application after every test class some files where remaining open.

File: app/controllers/TaxYearsController.scala [hunk 4]
```diff
@@ -17,49 +17,55 @@
...
+        isAgent = isAgent,
+        utr = user.saUtr,
+        itsaSubmissionIntegrationEnabled = isEnabled(ITSASubmissionIntegration),
+        earliestSubmissionTaxYear = earliestSubmissionTaxYear,
+        btaNavPartial = user.btaNavPartial,
+        origin = origin)
+    } match {
+      case Success(taxView) => Future.successful(Ok(taxView))
+      case Failure(_) =>
+        val errorHandler = if(isAgent) agentItvcErrorHandler else itvcErrorHandler
```
Comment (avelx) [inline app/controllers/TaxYearsController.scala:63]: if we can log error here

File: it/test/controllers/CalculationPollingControllerISpec.scala (hunk 3 above)
Comment (avelx) [reply it/test/controllers/CalculationPollingControllerISpec.scala:190]: Next story created to address this issue:
https://jira.tools.tax.service.gov.uk/browse/MISUV-8977

**Review by avelx — APPROVED** [review]
approved with comments

**Pull request merged**